app/models/user.rb

```diff
@@ -107,6 +107,11 @@ def inactive_message
     "Sorry, this account has been disabled."
   end
 
+  # Returns all users who match the query.
+  def self.search_from_query(members, query)
+    User.enabled.where.not(id: members).where(arel_table[:username].matches(query)).pluck(:username)
```

Comment:
I'd honestly go for option 2. I don't like options 3 and 4 because I think that scopes should only be created if we plan to use them in multiple places.